### PR TITLE
update get_interfaces

### PR DESCRIPTION
**Pull request by itdependsnetworks**
<!-- Make sure you have read http://napalm.readthedocs.io/en/latest/contributing/index.html --!>
Alternate to https://github.com/napalm-automation/napalm-ios/pull/151, pulled some code from @kbirkeland . 

Not yet ready to push, but wanted to put the PR in before I forgot :)

### CODE REVIEW

File: napalm_ios/ios.py [hunk 1]
```diff
@@ -919,91 +919,49 @@ def get_interfaces(self):
...
-        for interface in interface_list:
-            show_command = "show interface {0}".format(interface)
-            interface_output = self._send_command(show_command)
-            try:
-                # description filter
-                description = re.search(r"  Description: (.+)", interface_output)
-                interface_list[interface]['description'] = description.group(1).strip('\r')
-            except AttributeError:
-                interface_list[interface]['description'] = u'N/A'
+        interface = is_enabled = is_up = description =  mac_address = speed = speedformat = ''
```
Comment (mirceaulinic) [inline napalm_ios/ios.py]: Wow.
If you really want to do this, at least take `is_up` and `speed` away and default them to bool, respectively int.

File: napalm_ios/ios.py [hunk 2]
```diff
@@ -919,91 +919,49 @@ def get_interfaces(self):
...
+            if re.search(mac_addr_regex, line):
+                mac_addr_match = re.search(mac_addr_regex, line)
+                mac_address = mac_addr_match.groups()[0]
+
+            descr_regex = "^\s+Description: (.+?)$"
+            if re.search(descr_regex, line):
+                descr_match = re.search(descr_regex, line)
+                description = descr_match.groups()[0]
+
+            speed_regex = r"^\s+MTU \d+.+ BW (\d+) ([KMG]?b)"
```
Comment (mirceaulinic) [inline napalm_ios/ios.py]: Maybe `\s+` instead of hardcoded spaces?

**Comment by mirceaulinic** [comment]
Oh my, you weren't kidding when you said you want to improve the stats by adding lots of mock data.
BTW: we count the number of PRs, not the lines added. :]

**Comment by itdependsnetworks** [comment]
lol, just happen to work out that way. Been meaning to put in issue to clean up multiple show int calls, and the mac one and had seen the issue already opened. Had demo data from ntc-show and figured there was a reason we used both, so included them both :)

**Comment by dbarrosop** [comment]
Hahaha! keep them coming!

**Pull request merged**